### PR TITLE
Midi

### PR DESCRIPTION
**Pull request by jpcima**
- changed MIDI UI to be more tolerable. Icon stolen from Qjackctl.
- added ability to close MIDI port after it was opened
- handled errors in such way that exceptions will not raise if there was an error in the underlying MIDI system

### CODE REVIEW

File: src/midi/midi_rtmidi.cpp [hunk 1]
```diff
@@ -37,18 +36,22 @@ void MidiInRt::close()
...
 }
 
-void MidiInRt::openVirtual()
+bool MidiInRt::openVirtual()
 {
     RtMidiIn *midiin = lazyInstance();
-    close();
+    m_midiin->closePort();
+    m_errorSignaled = false;
     midiin->openVirtualPort("MIDI input");
```
Comment (Wohlstand) [inline src/midi/midi_rtmidi.cpp]: Why not to call this port as "OPL3 Bank Editor MIDI Input"? As this "MIDI Input" pseudo-device was shown at me

**Comment by papiezak** [comment]
is it possible to make midi input and testing section less wide on windows? it looks too stretched up

**Comment by Wohlstand** [comment]
On Linux that possible as ALSA allows to run any daemon that will create a MIDI device on the fly. On Windows is required to install a MIDI driver which will create a MIDI port.

**Comment by papiezak** [comment]
i meant UI aspect but thanks for the info.

File: src/midi/midi_rtmidi.cpp (hunk 1 above)
Comment (jpcima) [reply src/midi/midi_rtmidi.cpp]: It's a redundant display in jack midi but overall you're right about it. I'll change it.

**Comment by jpcima** [comment]
I can't reproduce such a sizing problem on Windows 7. But I'll do this: I'll set horizontal spacers to smaller widths. This might have an effect on your side.

**Pull request merged**